### ISSUE TITLE
Folksonomy Engine loads CSS from CDNs

### ISSUE THREAD

**Issue by hangy** [openfoodfacts/openfoodfacts-server#9012]
https://github.com/openfoodfacts/openfoodfacts-server/blob/a0e358e7a79f55ed457d5676067e121ff7097416/html/js/folksonomy.js#L63-L65

![image](https://github.com/openfoodfacts/openfoodfacts-server/assets/87124/27cd9042-f9e5-4f87-8166-0ec7794ee5dd)

For ProductOpener itself, we usually try to reduce the dependence on 3rd party CDNs. `folksonomy.js` does the opposite by loading three files from three different CDNs.

1. jQuery UI is already bundled with ProductOpener
2. FontAwesome could be revisited, because ProductOpener now mostly seems to use Material Icons since the redesign
3. Bundle jqGrid, or replace it (if there's feature parity) with other grid already bundled?

**Comment by teolemon** [comment]
/*
 * The Open Food Facts web app already loads jquery-ui.css but it doesn't work properly with the "dialog" function.
 * We add the CSS this way so that the embedded, relatively linked images load correctly.
 * (Use //ajax... so that https or http is selected as appropriate to avoid "mixed content".)
 */

@import url("https://ajax.googleapis.com/ajax/libs/jqueryui/1.11.4/themes/redmond/jquery-ui.css");
@import url("https://rawgit.com/free-jqgrid/jqGrid/master/css/ui.jqgrid.css");
@import url("https://netdna.bootstrapcdn.com/font-awesome/4.4.0/css/font-awesome.min.css");

**Comment by MSGanesh23** [comment]
Hi,
I’m new to contributing to Oppia and I’m planning to start working on this issue.
I’ll post updates here as I make progress. Could you please guide me, @hangy ?

**Comment by hangy** [comment]
I think this may be redundant if the folksonomy frontend is rebuilt as a [web component](https://github.com/openfoodfacts/openfoodfacts-webcomponents/discussions/32)

**Comment by dharsan-devaraj** [comment]
Hi,
I’d like to work on this issue.
Before starting implementation, I want to clarify the expected direction:
1. Should the goal be to fully remove external CDN CSS and bundle everything locally?
2. Is jqGrid expected to stay, or is replacing/removing it acceptable?
3. Should this be postponed if the folksonomy frontend is going to be rebuilt as a web component?
I can start by auditing current dependencies and proposing a concrete plan if that helps.
Thanks!

**Comment by LakshuCodes** [comment]
Hi
Given the mention that the frontend might be rebuilt as a web component, is this issue still something we should actively work on?

If yes, I’d be happy to start by auditing the current CDN dependencies and proposing a cleanup plan.

**Comment by bhaskar21-7** [comment]
Hi , I have submitted a PR for this issue: https://github.com/openfoodfacts/openfoodfacts-server/pull/13349

Please let me know if any changes are required.
